lib/manager/pub/artifacts.ts

```diff
@@ -0,0 +1,91 @@
+import { TEMPORARY_ERROR } from '../../constants/error-messages';
+import { logger } from '../../logger';
+import { ExecOptions, exec } from '../../util/exec';
+import {
+  getSiblingFileName,
+  readLocalFile,
+  writeLocalFile,
+} from '../../util/fs';
+import type { UpdateArtifact, UpdateArtifactsResult } from '../types';
+
+async function pubUpdate(
+  pubspecPath: string,
+  newPackageFileContent: string
+): Promise<void> {
+  let cmd = '';
+  if (newPackageFileContent.includes('sdk: flutter')) {
+    cmd = 'flutter pub upgrade';
+  } else {
+    cmd = 'pub upgrade';
```

Comment:
```suggestion
    cmd = 'dart pub upgrade';
```

needs to be pefixed with  `darrt` since v2 of the sdk and buildpack for now ionly supports dart sdk v2